presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationContext.java

```diff
@@ -13,110 +13,220 @@
  */
 package com.facebook.presto.verifier.framework;
 
-import com.facebook.presto.verifier.event.DeterminismAnalysisRun;
+import com.facebook.presto.jdbc.QueryStats;
+import com.facebook.presto.verifier.event.DeterminismAnalysisDetails;
 import com.facebook.presto.verifier.event.QueryFailure;
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableSet;
+import com.facebook.presto.verifier.event.VerifierQueryEvent;
 
+import java.util.HashSet;
 import java.util.List;
 import java.util.Optional;
+import java.util.Set;
 
-import static com.facebook.presto.verifier.framework.LimitQueryDeterminismAnalysis.NOT_RUN;
 import static com.google.common.base.Preconditions.checkState;
 import static com.google.common.collect.ImmutableList.toImmutableList;
 import static java.util.Objects.requireNonNull;
 
 public class VerificationContext
```

Comment:
It's strange to me that this class is mutable and gets passed around and modified in several places.  Usually we try to limit the use of mutable objects in Presto, especially across different classes.  In my opinion the mutability here makes the lifecycle of each object confusing and makes the code more error prone (e.g. trying to update the same field 2x). The second commit seems like it's fixing a problem that's mostly caused by this being a mutable object.  And then later commits add even more things to this class, which I think just makes it more fragile. I think it would be better to break this class up into a few classes that can each be immutable. 

For example, you can separate out an object for keeping the data verification info of control and test checksum queries and build that in DataVerfication.verify(), and then have another object that has all the determinism info that.gets built in the DeterminismAnalyzer and have the LimitDeterminismAnalysis also return the queryId, so that part can be immutable to.

Reply:
@rschlussel Comments addressed.

Instead of expanding the mutable object `VerificationContext`, changes were made to make it immutable. Separate checksum-related mutable fields into `CheckQueryContext`, and use `DeterminismAnalysisDetails.Builder` to replace determinism analysis related mutable fields.